ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/TransactionCommitterSpec.scala

```diff
@@ -259,11 +260,59 @@ class TransactionCommitterSpec extends AnyWordSpec with Matchers with MockitoSug
     "always set blindingInfo" in {
       val context = createCommitContext(recordTime = None)
 
-      val actual = transactionCommitter.blind(context, aTransactionEntrySummary)
+      val builder = TransactionBuilder(TransactionVersion.VDev)
+      val cid = builder.newCid
+
+      val actual =
+        transactionCommitter.blind(
+          context,
+          DamlTransactionEntrySummary(txEntryWithDivulgedContract(builder, cid)),
+        )
+
+      actual match {
+        case StepContinue(partialResult) =>
+          val actualDivulgencesList =
+            partialResult.submission.getBlindingInfo.getDivulgencesList.asScala
+              .map(entry =>
+                entry.getContractId -> entry.getDivulgedToLocalPartiesList.asScala.toSet
+              )
+
+          actualDivulgencesList should contain theSameElementsAs Vector(
+            cid.coid -> Set("ChoiceObserver")
+          )
+
+          val actualDisclosureList =
+            partialResult.submission.getBlindingInfo.getDisclosuresList.asScala
+              .map(entry => entry.getNodeId -> entry.getDisclosedToLocalPartiesList.asScala.toSet)
+
+          actualDisclosureList should contain theSameElementsAs Vector(
+            "0" -> Set("Alice"),
+            "1" -> Set("Actor", "Alice", "ChoiceObserver"),
+          )
+        case StepStop(_) => fail()
+      }
+    }
+
+    "always set divulgedContracts" in {
+      val context = createCommitContext(recordTime = None)
+
+      val builder = TransactionBuilder(TransactionVersion.VDev)
+      val cid = builder.newCid
+
+      val actual =
+        transactionCommitter.blind(
+          context,
+          DamlTransactionEntrySummary(txEntryWithDivulgedContract(builder, cid)),
+        )
 
       actual match {
         case StepContinue(partialResult) =>
-          partialResult.submission.hasBlindingInfo shouldBe true
+          val actualDivulgedContractIds =
+            partialResult.submission.getDivulgedContractsList.asScala.map(contract =>
+              contract.getContractId
+            )
+          actualDivulgedContractIds should contain theSameElementsAs Vector(cid.coid)
```

Comment:
@tudor-da : AFAIS, there is no test that compares the content of the `divulged_contracts` field is there?

Reply:
Addressed in the refactoring (expected vs actual contract instances are checked)